### PR TITLE
Improve styling of job history table

### PR DESCRIPTION
**Pull request by DuMaM**
This plugin base on many different sub pages. 
I want to update style for job config history. 
This will set up direction where those tables should be converted.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

image

### CODE REVIEW

**Comment by DuMaM** [comment]
Current problems:
* Somehow page switching stopped working :( 
* I don't know how to add submission for newer button style. 
* Every second "restore" button does not showing up

**Comment by DuMaM** [comment]
I found problem connected with page switch. It detects submission when there is no url attached to it and then calculate diff File.

**Comment by DuMaM** [comment]
Ok I fixed button problems. Now they do not make any submission, so everything works as expected.
Whats left is a "restore button" and positioning.

**Comment by DuMaM** [comment]
image
Current progress:
* buttons positioning got fixed 
* Show Diff submission stopped working :/
* Button themselves should act as a hrefs
* Still I didn't found a reason why "restore" icon do not show it self

**Comment by DuMaM** [comment]
Current progress:
image

Current problems:
* `all` button do not highlight
* `restore` icon do not show itself for all config changes

**Comment by DuMaM** [comment]
Only restore button problem left.

**Comment by DuMaM** [comment]
image
Everything now works 😉

**Comment by NotMyFault** [comment]
Looks very nice, thanks for your contribution!

I'm wondering what's up with the radio buttons, they seem to be not aligned in the table center and move around if you click on them:
https://user-images.githubusercontent.com/13383509/172365618-2522a607-47fd-40bb-af86-1088d7070a7d.mp4

**Comment by NotMyFault** [comment]
Looks very nice, thanks for your contribution!

I'm wondering what's up with the radio buttons, they seem to be not aligned in the table center and move around if you click on them:
https://user-images.githubusercontent.com/13383509/172365618-2522a607-47fd-40bb-af86-1088d7070a7d.mp4

**Comment by NotMyFault** [comment]
> > I'm wondering what's up with the radio buttons, they seem to be not aligned in the table center and move around if you click on them: https://user-images.githubusercontent.com/13383509/172365618-2522a607-47fd-40bb-af86-1088d7070a7d.mp4
> 
> Thats the case for all radios in this jenkins. I got similar experience in other plugins.

Oic, that should probably be reported on Jira as core issue, if it affects several plugins.

File: src/main/resources/hudson/plugins/jobConfigHistory/JobConfigHistoryProjectAction/index.jelly [hunk 1]
```diff
@@ -123,24 +115,27 @@
...
-                                <button type="button" class="jch delete-button" onClick="removeEntryFromTable('table-row-${configNr}', '${config.date}', 'null', '${message}')" value="X">
-                                  <l:icon class="icon-stop icon-sm" alt="${%Delete Revision}"/>
-                                </button>
-                            </td>
-                          </j:if>
+                          <!-- Trash icon -->
+                          <td style="text-align:center">
+                            <j:if test="${it.hasDeleteEntryPermission()}">
+                              <j:set var="message" value="${%Do you really want to delete the history entry} "/>
+                              <button type="button" class="jenkins-button jenkins-button--transparent" onClick="removeEntryFromTable('table-row-${configNr}', '${config.date}', 'null', '${message}')" value="X">
```
Comment (NotMyFault) [inline src/main/resources/hudson/plugins/jobConfigHistory/JobConfigHistoryProjectAction/index.jelly:134]:
```suggestion
                              <button type="button" class="jenkins-button jenkins-button--destructive" onClick="removeEntryFromTable('table-row-${configNr}', '${config.date}', 'null', '${message}')" value="X">

```
A red button may hightlight a destructive action better.

Comment (DuMaM) [reply src/main/resources/hudson/plugins/jobConfigHistory/JobConfigHistoryProjectAction/index.jelly]: Good idea :)

**Comment by DuMaM** [comment]
@NotMyFault Is it ok to use transparent button for '<' '>' when they are inactive?
Maybe normal one will be better?

**Comment by NotMyFault** [comment]
> @NotMyFault Is it ok to use transparent button for '<' '>' when they are inactive?

Where do you mean exactly?

**Comment by DuMaM** [comment]
> > @NotMyFault Is it ok to use transparent button for '<' '>' when they are inactive?
> 
> Where do you mean exactly?

Inactive (no page 0)
image

Active (there is previous page -> 1)
image

**Comment by NotMyFault** [comment]
Oic, yeah I agree, the normal button looks better.

**Comment by DuMaM** [comment]
Is there a way to show inactive button? If i use a normal one then people may get confused

**Comment by NotMyFault** [comment]
> Is there a way to show inactive button? If i use a normal one then people may get confused

The normal one as seen on

looks like any other modern button in Jenkins.

I'm unsure if choosing a different style might break with this scheme 🤔

**Comment by DuMaM** [comment]
image
I'm pretty happy with current result :)

**Comment by DuMaM** [comment]
@NotMyFault I fixed urls for Gitpod now. At least for pages I was working with.
Also I updated one more site (root one).
image
There was duplicated filter functionality implemented as urls in `Main-panel` buttons and `Side-panels` tasks, so i removed those main ones.

**Comment by NotMyFault** [comment]
> @NotMyFault I fixed urls for Gitpod now. At least for pages I was working with.

Yup, all looking fine so far 👏🏻 

Though the table headers aren't set and follow your scroll behavior:
https://user-images.githubusercontent.com/13383509/173225019-67294d79-1d91-4d1e-a9c9-00de0461ed83.mp4

**Comment by DuMaM** [comment]
> > @NotMyFault I fixed urls for Gitpod now. At least for pages I was working with.
> 
> Yup, all looking fine so far 👏🏻
> 
> Though the table headers aren't set and follow your scroll behavior: https://user-images.githubusercontent.com/13383509/173225019-67294d79-1d91-4d1e-a9c9-00de0461ed83.mp4

Headers fixed :)

**Review by NotMyFault — APPROVED** [review]
Thanks a bunch for your contribution, nice to see the modernized jenkins tables here as well (at least at this specific history view).

**Pull request merged**